### PR TITLE
New package: znc

### PR DESCRIPTION
**Pull request by Yonle**
ZNC is an advanced IRC Bouncer.

### CODE REVIEW

**Comment by Yonle** [comment]
For unknown reasons, The configure script address perl to `/usr/bin/perl`

```
2022-02-21T11:50:28.6860908Z configure:5328: checking for perl
2022-02-21T11:50:28.6861130Z configure:5346: found /usr/bin/perl
2022-02-21T11:50:28.6861366Z configure:5358: result: /usr/bin/perl
2022-02-21T11:50:28.6861761Z configure:5369: checking for perl_alloc in -lperl
2022-02-21T11:50:28.6863990Z configure:5394: aarch64-linux-android-clang++ -o conftest  -fstack-protector-strong -Oz  -O2 -D_FORTIFY_SOURCE=2 -Wall -W -Wno-unused-parameter -Woverloaded-virtual -Wshadow  -I/data/data/com.termux/files/usr/include -I/data/data/com.termux/files/usr/include  -I/data/data/com.termux/files/usr/include -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -fopenmp -static-openmp -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now -Wl,--export-dynamic -Wl,-E -fstack-protector-strong -L/usr/local/lib -L/usr/lib/x86_64-linux-gnu/perl/5.32/CORE -lperl -ldl -lm -lpthread -lc -lcrypt -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/lib/x86_64-linux-gnu/perl/5.32/CORE conftest.cpp -lperl  -L/data/data/com.termux/files/usr/lib -lssl -lcrypto -lz -L/data/data/com.termux/files/usr/lib -licuuc -licudata >&5
2022-02-21T11:50:28.6865445Z ld: error: unable to find library -lperl
2022-02-21T11:50:28.6865781Z ld: error: unable to find library -lpthread
2022-02-21T11:50:28.6866089Z ld: error: unable to find library -lperl
2022-02-21T11:50:28.6866498Z clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Yonle** [comment]
And i'm forget to add perl as dependencies! Damn!

**Comment by Yonle** [comment]
Ready to go now. Tested and ZNC worked well as expected.

File: packages/znc/no_pthread.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+--- znc-1.8.2/src/main.cpp	2020-09-08 05:57:50.000000000 +0700
++++ znc-1.8.2.mod/src/main.cpp	2022-02-21 17:08:57.974706825 +0700
+@@ -180,7 +180,6 @@
+         m_thread = std::thread([=]() { HandleSignals(pZNC); });
+     }
+     ~CSignalHandler() {
+-        pthread_cancel(m_thread.native_handle());
```
Comment (xtkoba) [inline packages/znc/no_pthread.patch:7]: Please justify the removal of `pthread_cancel()` call. Doesn't this cause hang up in any cases? Or do we need some workaround as in the `calcurse` package?

Comment (Yonle) [reply packages/znc/no_pthread.patch:7]: Thinking about it. Maybe i will do some workaround as in `calcurse` package.

**Pull request closed without merging**